eng/PrepareRelease.yml

```diff
@@ -2,15 +2,43 @@ stages:
 - stage: PrepareReleaseStage
   displayName: Release Preparation
   dependsOn:
-  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')) }}:
+  - ${{ if 
+      or(
+        and(
+          ne(variables['System.TeamProject'], 'public'), 
+          notin(variables['Build.Reason'], 'PullRequest'), 
+          or(
+            startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), 
+            startsWith(variables['Build.SourceBranch'], 'refs/heads/internal/release/')
+          )
+        ),
+        and(
+          ne(variables['System.TeamProject'], 'public'), 
+          startsWith(variables['Build.SourceBranch'], 'refs/heads/test/release/')
+        )
+      ) }}:
     - publish_using_darc
   jobs:
   - job: PrepareReleaseJob
     displayName: Prepare release with Darc
     pool: 
       vmImage: windows-latest
     variables:
-    - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')) }}:
+    - ${{ if 
+        or(
+          and(
+            ne(variables['System.TeamProject'], 'public'), 
+            notin(variables['Build.Reason'], 'PullRequest'), 
+            or(
+              startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), 
+              startsWith(variables['Build.SourceBranch'], 'refs/heads/internal/release/')
+            )
+          ),
+          and(
+            ne(variables['System.TeamProject'], 'public'), 
+            startsWith(variables['Build.SourceBranch'], 'refs/heads/test/release/')

```

Comment:
Can you simplify this by moving this branch check to be inline with the other ones? I don't think we need to allow internal PR builds to attempt these steps; just invoke the build directly on the branch if need be.

Reply:
Do the same for all of the other conditionals you've updated.

Reply:
If I understand your request correctly, I can't do that. The reason this is separated out is because we need to allow build on PR (since that's the whole purpose of the branch).